ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/vm/VmTraceGenerator.java

```diff
@@ -97,7 +98,10 @@ private void addFrame(final TraceFrame frame) {
   private boolean mustIgnore(final TraceFrame frame) {
     if ("STOP".equals(frame.getOpcode()) && transactionTrace.getTraceFrames().size() == 1) {
       return true;
-    } else if (!frame.getExceptionalHaltReasons().isEmpty()) {
+    } else if (!frame.getExceptionalHaltReasons().isEmpty()
+        && !frame
+            .getExceptionalHaltReasons()
+            .contains(ExceptionalHaltReason.INVALID_JUMP_DESTINATION)) {
```

Comment:
Curious, does Parity add frames for all exceptional halt cases except invalid jumps?  Or do we have disjoint set of halt reasons?  i.e. Parity code is fundamentally different in this area.